sdk/analytics/src/main/java/com/klaviyo/analytics/networking/requests/KlaviyoApiRequest.kt

```diff
@@ -202,17 +206,20 @@ internal open class KlaviyoApiRequest(
      *
      * @returns The string value of the response body, if one was returned
      */
-    fun send(): Status {
+    fun send(beforeSend: (() -> Unit)? = null): Status {
         if (!Registry.networkMonitor.isNetworkConnected()) {
```

Comment:
Just curious: is this based on notifications from the network listener or is it asking the network if it's connected. If the latter might want to see how this affects battery usage.

Reply:
Good question, I believe this spot check is asking the system directly. I can probably optimize that in the network service a bit